taesch/lib/view/screen/shops_map_screen.dart

```diff
@@ -20,24 +20,144 @@ class ShopsMapScreen extends StatefulWidget {
 
 class _ShopsMapScreenState extends State<ShopsMapScreen> {
   LoggerWrapper logger = LoggerWrapper();
+  int _onTapId = 0;
 
   List<Marker> _getMarkersFromSpot(List<MapSpot> mapSpots) {
     List<Marker> markers = [];
+    int idCounter=0;
     for (MapSpot mapSpot in mapSpots) {
+      idCounter++;
+      int markerID = idCounter;
+      Text markerDescriptor = const Text("");
+
+      if (_onTapId == markerID){
+        markerDescriptor = Text(
+            mapSpot.name, //\n${mapSpot.address}
+            style: const TextStyle(fontWeight: FontWeight.bold)
+        );
+      }else{
+        //markerDescriptor = const Text("");
+      }
+
       markers.add(Marker(
+          width: 200,
+          height: 300,
+          key: ValueKey(markerID),
           point: LatLng(mapSpot.lat, mapSpot.long),
-          /*builder: (ctx) => GestureDetector(
-                        onTap: () {
-                            /*ScaffoldMessenger.of(ctx).showSnackBar(const SnackBar(
-                                content: Text('Tapped on blue FlutterLogo Marker'),
-                                )
-                            );*/
-                            print("Tapped on marker.");
-                        },
-                        child: Icon(Icons.location_on),
+          builder: (ctx) => GestureDetector(
+            onTap: () {
+              setState(() {
+                // show information only for this marker
+                _onTapId = markerID;
+              });
+            },
+
+            onLongPress: (){
 
-                    )*/
-          builder: (ctx) => const Icon(Icons.location_on)));
+              if (_onTapId!=markerID){
+                // gleiches Verhalten wie onTap()
+                setState(() {
+                  _onTapId = markerID;
+                });
+              }
+
+              showDialog(
```

Comment:
Bitte dieses Dialogfenster in eine eigene Klasse auslagern nach custom_widgets.

Reply:
gemacht